test/handshake_helper.c

```diff
@@ -144,6 +188,33 @@ static int servername_reject_cb(SSL *s, int *ad, void *arg)
     return select_server_ctx(s, arg, 0);
 }
 
+static int early_ignore_cb(SSL *s, SSL_EARLY *ectx, int *al, void *arg)
+{
+    if (!early_select_server_ctx(s, ectx, arg, 1)) {
+        *al = SSL_AD_UNRECOGNIZED_NAME;
+        return 0;
+    }
+    return 1;
+}
+
+static int early_reject_cb(SSL *s, SSL_EARLY *ectx, int *al, void *arg)
+{
+    if (!early_select_server_ctx(s, ectx, arg, 0)) {
+        *al = SSL_AD_UNRECOGNIZED_NAME;
+        return 0;
+    }
+    return 1;
+}
+
+static int early_nov12_cb(SSL *s, SSL_EARLY *ectx, int *al, void *arg)
+{
+    int ret = early_select_server_ctx(s, ectx, arg, 0);
+    SSL_set_max_proto_version(s, TLS1_1_VERSION);
+    if (!ret)
```

Comment:
I guess we could add some calls to the SSL_EARLY_* accessors in here to stop coveralls from complaining.
(And please confirm that I should make the change to putting the SSL_EARLY object first, or make a new type for it to avoid the need to pass  the SSL as well, or ...)